build-calico-resource.sh

```diff
@@ -1,22 +1,50 @@
-#!/bin/sh
+#!/bin/bash
 set -eux
 
-rm -rf resource-build
-mkdir resource-build
-cd resource-build
-wget https://github.com/projectcalico/calicoctl/releases/download/v1.6.4/calicoctl
-wget https://github.com/projectcalico/cni-plugin/releases/download/v1.11.6/calico
-wget https://github.com/projectcalico/cni-plugin/releases/download/v1.11.6/calico-ipam
-chmod +x calicoctl calico calico-ipam
-
-mkdir temp
-(cd temp
-  wget https://github.com/containernetworking/plugins/releases/download/v0.7.1/cni-plugins-amd64-v0.7.1.tgz
-  tar -vxf cni-plugins-amd64-v0.7.1.tgz
-  mv portmap ..
-)
-rm -rf temp
-
-tar -vcaf ../calico-resource.tar.gz .
-cd ..
-rm -rf resource-build
+# Supported calico architectures
+arches="amd64 arm64"
+
+# 2.6.x has no binary releases for arm64; fetch them from neander
+neander="ubuntu@10.96.66.14"
+
+fetch_cni_plugins() {
+  arch=${1:-}
+  if [ -z ${arch} ]; then
+    echo "Missing arch parameter to fetch_cni_plugins"
+    exit 1
+  fi
+
+  mkdir temp
+  (cd temp
+    wget https://github.com/containernetworking/plugins/releases/download/v0.7.1/cni-plugins-${arch}-v0.7.1.tgz
+    tar -vxf cni-plugins-${arch}-v0.7.1.tgz
+    mv portmap ..
+  )
+  rm -rf temp
+}
+
+for arch in ${arches}; do
+  rm -rf resource-build-$arch
+  mkdir resource-build-$arch
+  pushd resource-build-$arch
+
+  if [ $arch = "amd64" ]; then
+    wget https://github.com/projectcalico/calicoctl/releases/download/v1.6.4/calicoctl
+    wget https://github.com/projectcalico/cni-plugin/releases/download/v1.11.6/calico
+    wget https://github.com/projectcalico/cni-plugin/releases/download/v1.11.6/calico-ipam
+  elif [ $arch = "arm64" ]; then
+    scp ${neander}:~/go/src/github.com/projectcalico/calicoctl/dist/calicoctl-linux-arm64 ./calicoctl
+    scp ${neander}:~/go/src/github.com/projectcalico/cni-plugin/dist/calico ./calico
+    scp ${neander}:~/go/src/github.com/projectcalico/cni-plugin/dist/calico-ipam ./calico-ipam
```

Comment:
Were the arm64 bins relatively straightforward to build? If neander blows up and you're on vacation, will someone else on the team be able to rebuild these easily enough?

I really want to say we should have every step in this process automated, including building the arm64 resources, but I guess it doesn't make sense to spend time on that when it'd be about the same effort to just do the damn upgrade to Calico 3.x.

Do we have anywhere to store these bins that's more robust? We could throw them on S3 or something but that seems ridiculous. It'd be nice if we had something internal for this.

To be clear, I'm not asking for any changes here. Just fishing for any low-effort ways to make this setup a little more robust.

Reply:
> Were the arm64 bins relatively straightforward to build? If neander blows up and you're on vacation, will someone else on the team be able to rebuild these easily enough?

Notes on building the bins are in [this gist](https://gist.github.com/kwmonroe/9b5f8dac2c17f93629a1a3868b22d671). This should make its way to a wiki i guess.

> do the damn upgrade to Calico 3.x.

+1

> Do we have anywhere to store these bins that's more robust?

Good call, especially since the neander location required you to be on the vpn.  I've moved them to:

https://people.canonical.com/~kwmonroe/calico-2.6-arm64/